### DISCUSSION TITLE
pip install  paddleocr  err  help!!!

### DISCUSSION THREAD

**Discussion by Lucky-lhq** [PaddlePaddle/PaddleOCR#12248] [Q&A] [answered]
Building wheels for collected packages: PyMuPDF
  Building wheel for PyMuPDF (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [79 lines of output]
      PyMuPDF/setup.py: sys.argv: ['C:\\Users\\KF\\AppData\\Local\\Temp\\pip-install-22zlm_ab\\pymupdf_63196f9920f14373802ec1b56c5ce651\\setup.py', 'bdist_wheel', '-d', 'C:\\Users\\KF\\AppData\\Local\\Temp\\pip-wheel-fymf1p00']     
      PyMuPDF/setup.py: os.getcwd(): C:\Users\KF\AppData\Local\Temp\pip-install-22zlm_ab\pymupdf_63196f9920f14373802ec1b56c5ce651
      PyMuPDF/setup.py: __file__: C:\Users\KF\AppData\Local\Temp\pip-install-22zlm_ab\pymupdf_63196f9920f14373802ec1b56c5ce651\setup.py
      PyMuPDF/setup.py: $PYTHON_ARCH: None
      PyMuPDF/setup.py: os.environ (51):
      PyMuPDF/setup.py:     PSEXECUTIONPOLICYPREFERENCE: RemoteSigned
      PyMuPDF/setup.py:     TERM_SESSION_ID: 5a35fc27-5a86-4020-8558-07048a28995c
      PyMuPDF/setup.py:     VIRTUAL_ENV: D:\Env\pdf_image
      PyMuPDF/setup.py:     ZES_ENABLE_SYSMAN: 1
      PyMuPDF/setup.py:     PROGRAMW6432: C:\Program Files
      PyMuPDF/setup.py:     PYCHARM: C:\Program Files\JetBrains\PyCharm 2022.3.2\bin;
      PyMuPDF/setup.py:     COMMONPROGRAMW6432: C:\Program Files\Common Files
      PyMuPDF/setup.py:     FIG_JETBRAINS_SHELL_INTEGRATION: 1
      PyMuPDF/setup.py:     USERNAME: KF
      PyMuPDF/setup.py:     USERPROFILE: C:\Users\KF
      PyMuPDF/setup.py:     ALLUSERSPROFILE: C:\ProgramData
      PyMuPDF/setup.py:     PROCESSOR_REVISION: 5e03
      PyMuPDF/setup.py:     IDEA_INITIAL_DIRECTORY: C:\Users\KF\Desktop
      PyMuPDF/setup.py:     PUBLIC: C:\Users\Public
      PyMuPDF/setup.py:     PATH: D:\Env\pdf_image/Scripts;C:\Users\KF\AppData\Local\Programs\Python\Python311\Scripts\;C:\Users\KF\AppData\Local\Programs\Python\Python311\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\
Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Mic
rosoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\ManagementStudio\;C:\Program Files (x86)\Microsoft SQL Server\100\Tools\Binn\;C:\Program Files\Microsoft SQL Server\100\Tools\Binn\;C:\Progra
m Files\Microsoft SQL Server\100\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\100\Tools\Binn\VSShell\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio 9.0\Common7\IDE\PrivateAssemblies\;C:\Program Files (x86)\Micr
osoft SQL Server\100\DTS\Binn\;E:\MySql\MySQL Server 5.7\bin;C:\ProgramData\chocolatey\bin;E:\NodeJs\;E:\NodeJs\node_global\node_modules;C:\Program Files\d;C:\Program Files\Git\cmd;C:\Program Files\Redis\;C:\Users\KF\AppData\Local\M
icrosoft\WindowsApps;C:\Users\KF\AppData\Local\Programs\Microsoft VS Code\bin;C:\Program Files\JetBrains\PyCharm 2022.3.2\bin;;E:\NodeJs\node_global;C:\Users\KF\.dotnet\tools
      PyMuPDF/setup.py:     DRIVERDATA: C:\Windows\System32\Drivers\DriverData
      PyMuPDF/setup.py:     HOMEDRIVE: C:
      PyMuPDF/setup.py:     SESSIONNAME: Console
      PyMuPDF/setup.py:     TERMINAL_EMULATOR: JetBrains-JediTerm
      PyMuPDF/setup.py:     LOGONSERVER: \\LHQ-PC
      PyMuPDF/setup.py:     SYSTEMROOT: C:\Windows
      PyMuPDF/setup.py:     HOMEPATH: \Users\KF
      PyMuPDF/setup.py:     CHOCOLATEYINSTALL: C:\ProgramData\chocolatey
      PyMuPDF/setup.py:     LOCALAPPDATA: C:\Users\KF\AppData\Local
      PyMuPDF/setup.py:     APPDATA: C:\Users\KF\AppData\Roaming
      PyMuPDF/setup.py:     PROCESSOR_IDENTIFIER: Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
      PyMuPDF/setup.py:     PATHEXT: .COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW;.CPL
      PyMuPDF/setup.py:     PSMODULEPATH: C:\Users\KF\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules;C:\Program Files (x86)\Microsoft SQL Server\130\
Tools\PowerShell\Modules\
      PyMuPDF/setup.py:     PROGRAMFILES(X86): C:\Program Files (x86)
      PyMuPDF/setup.py:     OS: Windows_NT
      PyMuPDF/setup.py:     PROCESSOR_ARCHITECTURE: AMD64
      PyMuPDF/setup.py:     NUMBER_OF_PROCESSORS: 4
      PyMuPDF/setup.py:     COMSPEC: C:\Windows\system32\cmd.exe
      PyMuPDF/setup.py:     WORKON_HOME: D:\Env
      PyMuPDF/setup.py:     PROCESSOR_LEVEL: 6
      PyMuPDF/setup.py:     WINDIR: C:\Windows
      PyMuPDF/setup.py:     USERDOMAIN_ROAMINGPROFILE: LHQ-PC
      PyMuPDF/setup.py:     PROGRAMFILES: C:\Program Files
      PyMuPDF/setup.py:     TMP: C:\Users\KF\AppData\Local\Temp
      PyMuPDF/setup.py:     TEMP: C:\Users\KF\AppData\Local\Temp
      PyMuPDF/setup.py:     COMMONPROGRAMFILES(X86): C:\Program Files (x86)\Common Files
      PyMuPDF/setup.py:     ONEDRIVE: C:\Users\KF\OneDrive
      PyMuPDF/setup.py:     ONEDRIVECONSUMER: C:\Users\KF\OneDrive
      PyMuPDF/setup.py:     USERDOMAIN: LHQ-PC
      PyMuPDF/setup.py:     SYSTEMDRIVE: C:
      PyMuPDF/setup.py:     COMPUTERNAME: LHQ-PC
      PyMuPDF/setup.py:     PROGRAMDATA: C:\ProgramData
      PyMuPDF/setup.py:     CHOCOLATEYLASTPATHUPDATE: 133267971471785433
      PyMuPDF/setup.py:     NODE_PATH: E:\NodeJs\node_global\node_modules
      PyMuPDF/setup.py:     COMMONPROGRAMFILES: C:\Program Files\Common Files
      PyMuPDF/setup.py:     PIP_BUILD_TRACKER: C:\Users\KF\AppData\Local\Temp\pip-build-tracker-kdd_oc5d
      PyMuPDF/setup.py: mupdf_tgz already exists: C:\Users\KF\AppData\Local\Temp\pip-install-22zlm_ab\pymupdf_63196f9920f14373802ec1b56c5ce651\mupdf.tgz
      PyMuPDF/setup.py: Extracting C:\Users\KF\AppData\Local\Temp\pip-install-22zlm_ab\pymupdf_63196f9920f14373802ec1b56c5ce651\mupdf.tgz
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\KF\AppData\Local\Temp\pip-install-22zlm_ab\pymupdf_63196f9920f14373802ec1b56c5ce651\setup.py", line 487, in <module>
          mupdf_local = get_mupdf()
                        ^^^^^^^^^^^
        File "C:\Users\KF\AppData\Local\Temp\pip-install-22zlm_ab\pymupdf_63196f9920f14373802ec1b56c5ce651\setup.py", line 450, in get_mupdf
          return tar_extract( mupdf_tgz, exists='return')
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\KF\AppData\Local\Temp\pip-install-22zlm_ab\pymupdf_63196f9920f14373802ec1b56c5ce651\setup.py", line 183, in tar_extract
          t.extractall()
        File "C:\Users\KF\AppData\Local\Programs\Python\Python311\Lib\tarfile.py", line 2059, in extractall
          self.extract(tarinfo, path, set_attrs=not tarinfo.isdir(),
        File "C:\Users\KF\AppData\Local\Programs\Python\Python311\Lib\tarfile.py", line 2100, in extract
          self._extract_member(tarinfo, os.path.join(path, tarinfo.name),
        File "C:\Users\KF\AppData\Local\Programs\Python\Python311\Lib\tarfile.py", line 2173, in _extract_member
          self.makefile(tarinfo, targetpath)
        File "C:\Users\KF\AppData\Local\Programs\Python\Python311\Lib\tarfile.py", line 2214, in makefile
          with bltn_open(targetpath, "wb") as target:
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
      FileNotFoundError: [Errno 2] No such file or directory: '.\\mupdf-1.20.3-source\\thirdparty\\harfbuzz\\test\\shaping\\texts\\in-house\\shaper-indic\\script-devanagari\\utrrs\\codepoint\\IndicFontFeatureCodepoint-AdditionalCons
onants.txt'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for PyMuPDF
  Running setup.py clean for PyMuPDF
Failed to build PyMuPDF
ERROR: Could not build wheels for PyMuPDF, which is required to install pyproject.toml-based projects

**Comment by chenhao86** [comment, marked as answer]
用这个版本试试。直接 pip install D:\PyMuPDF-1.20.2-cp311-cp311-win_amd64.whl
[PyMuPDF-1.20.2-cp311-cp311-win_amd64.whl](https://drive.google.com/file/d/1bYdtLr2tkyWZJr-TBSO8MB6w-BZTxkaG/view?usp=drive_link)

**Comment by lithStudy** [comment]
https://github.com/PaddlePaddle/PaddleOCR/issues/10780#issuecomment-1700256944
有效，感谢
